src/Razor/src/Microsoft.CodeAnalysis.Remote.Razor/DevTools/RemoteDevToolsService.cs

```diff
@@ -0,0 +1,126 @@
+        DocumentId razorDocumentId,
+        CancellationToken cancellationToken)
+        => RunServiceAsync(
+            solutionInfo,
+            razorDocumentId,
+            context => GetCSharpDocumentTextAsync(context, cancellationToken),
+            cancellationToken);
+
+    public ValueTask<string?> GetHtmlDocumentTextAsync(
+        RazorPinnedSolutionInfoWrapper solutionInfo,
+        DocumentId razorDocumentId,
+        CancellationToken cancellationToken)
+        => RunServiceAsync(
+            solutionInfo,
+            razorDocumentId,
+            context => GetHtmlDocumentTextAsync(context, cancellationToken),
+            cancellationToken);
+
+    public ValueTask<string?> GetFormattingDocumentTextAsync(
+        RazorPinnedSolutionInfoWrapper solutionInfo,
+        DocumentId razorDocumentId,
+        CancellationToken cancellationToken)
+        => RunServiceAsync(
+            solutionInfo,
+            razorDocumentId,
+            context => GetFormattingDocumentTextAsync(context, cancellationToken),
+            cancellationToken);
+
+    public ValueTask<string> GetTagHelpersJsonAsync(
+        RazorPinnedSolutionInfoWrapper solutionInfo,
+        DocumentId razorDocumentId,
+        CancellationToken cancellationToken)
+        => RunServiceAsync(
+            solutionInfo,
+            razorDocumentId,
+            context => GetTagHelpersJsonAsync(context, cancellationToken),
+            cancellationToken);
+
+    public ValueTask<Microsoft.CodeAnalysis.Razor.Protocol.DevTools.RazorSyntaxTree?> GetRazorSyntaxTreeAsync(
+        RazorPinnedSolutionInfoWrapper solutionInfo,
+        DocumentId razorDocumentId,
+        CancellationToken cancellationToken)
+        => RunServiceAsync(
+            solutionInfo,
+            razorDocumentId,
+            context => GetRazorSyntaxTreeAsync(context, cancellationToken),
+            cancellationToken);
+
+    private async ValueTask<string?> GetCSharpDocumentTextAsync(RemoteDocumentContext documentContext, CancellationToken cancellationToken)
+    {
+        var codeDocument = await documentContext.GetCodeDocumentAsync(cancellationToken).ConfigureAwait(false);
+        return codeDocument.GetCSharpSourceText().ToString();
+    }
+
+    private async ValueTask<string?> GetHtmlDocumentTextAsync(RemoteDocumentContext documentContext, CancellationToken cancellationToken)
+    {
+        var codeDocument = await documentContext.GetCodeDocumentAsync(cancellationToken).ConfigureAwait(false);
+        return codeDocument.GetHtmlSourceText().ToString();
+    }
+
+    private async ValueTask<string?> GetFormattingDocumentTextAsync(RemoteDocumentContext documentContext, CancellationToken cancellationToken)
+    {
+        // For formatting, we typically want the C# generated document
+        var codeDocument = await documentContext.GetCodeDocumentAsync(cancellationToken).ConfigureAwait(false);
+        return codeDocument.GetCSharpSourceText().ToString();
```

Comment:
This should be return `CSharpFormattingPass.GetFormattingDocumentContentsForSyntaxVisualizer(codeDocument);`, with the appropriate using statement added at the top of the file, and `#pragma warning disable` for the obsolete warnings, around the call.